src/App/Fossa/YamlDeps.hs

```diff
@@ -78,32 +78,30 @@ toSourceUnit root yamldeps@YamlDependencies{..} = do
       , additionalData = additional
       }
 
-toBuildData :: NE.NonEmpty ReferencedDependency -> SourceUnitBuild
-toBuildData deps =
+toBuildData :: NE.NonEmpty Locator -> SourceUnitBuild
```

Comment:
[no action]: For now, I'm okay with this.  Long-term, I prefer to let internal models exist as raw/unprocessed as possible until they are needed, and I think `Locator` is not the best thing to use in this position.

For example, I would prefer passing back some structure that could be turned into a locator, e.g. `Archive` and `arcToLocator`.  This ensures that we don't limit ourselves to our output representation, and instead retain full flexibility of our own internal data model.

I think in this case, there's no real concern, but it is a pattern I'd like to see us do more of.

Reply:
Yea I agree with you, however in this case returning something other than Locator requires another layer of transformation which feels overkill.

Reply:
Like I said, for now this is perfectly fine.  I think we may revisit this when we deprecate srclib.